NDTensors/src/lib/Sectors/src/category_product.jl

```diff
@@ -8,22 +15,32 @@ CategoryProduct(c::CategoryProduct) = _CategoryProduct(categories(c))
 
 categories(s::CategoryProduct) = s.cats
 
-Base.isempty(S::CategoryProduct) = isempty(categories(S))
-Base.length(S::CategoryProduct) = length(categories(S))
-Base.getindex(S::CategoryProduct, args...) = getindex(categories(S), args...)
+const TrivialSector = CategoryProduct{Tuple{}}
```

Comment:
At this point I only see `TrivialSector` used for a few `fusion_rule` definitions, can we get rid of it completely? Also I would prefer to try to remove those `fusion_rule` definitions that are hardcoded to handle the empty limit and just rely on the generic `CategoryProduct` fusion code.

Reply:
My current suggestion is to introduce a devoted type `struct TrivialSector <: AbstractCategory end` to serve as a universal trivial sector, and no long treat `CategoryProduct{Tuple{}}` as special in any way, so I'll close this comment.